### ISSUE TITLE
Advice on embedding dotspatial within a larger application

### ISSUE THREAD

**Issue by snahdog** [DotSpatial/DotSpatial#874]
I have a larger application with multiple different functionalities. One of those functionalities uses dotspatial to display and manipulate a map. My idea order of operations would be to:
1. Open up a form with a menustrip with multiple items.
2. Upon clicking on one items of the menustrip, a dotspatial map control and legend would be made available to the user, in addition to some toolbars (pan, zoom, etc.)
3. Upon clicking other items, the map (and other dospatial stuff) should be disposed  and a completely new set of controls would be made available to the other.

I have been using the demomap to launch dotspatial as part of the startup form, but i am having trouble making the transition to a scenario where dotspatial is not loaded on the startup form. One possibility that i have been exploring is to keep my original dotspatial form and just open it up inside of the main form of my new application. However, if possible, i'd rather treat is as a user control. Unfortunately, I get all kinds of weird stuff happening like menubars popping out of nowhere, Iheader error messages, and the extensions not loading.

I would appreciate guidance/advice on how to structure an application like this. Thanks.

**Comment by mogikanin** [comment]
@snahdog do you still have troubles with it? :)

**Comment by snahdog** [comment]
I could not figure out how to make a user control and have things work correctly. What I did instead was the alternative that I described in my initial post: I have a main application form and I use it to open up a form similar to the demomap application inside of that main form and set the form characteristics in a way that it looks and functions like a user control. It works fine in that case. The downside is that each time I open up a form like this, it has to reload the extensions, which takes a bit more time.
